src/xamarin/sk_managed_png_chunk_reader.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 Microsoft Corporation. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include "include/xamarin/SkManaged_Png_Chunk_Reader.h"
+
+#include "include/xamarin/sk_managed_png_chunk_reader.h"
+#include "src/c/sk_types_priv.h"
+
+static inline SkManaged_Png_Chunk_Reader* AsManaged_Png_Chunk_Reader(sk_managed_png_chunk_reader_t* d) {
+    return reinterpret_cast<SkManaged_Png_Chunk_Reader*>(d);
+}
+static inline sk_managed_png_chunk_reader_t* ToManaged_Png_Chunk_Reader(SkManaged_Png_Chunk_Reader* d) {
+    return reinterpret_cast<sk_managed_png_chunk_reader_t*>(d);
+}
+
+static sk_managed_png_chunk_reader_procs_t gProcs;
+
+bool readChunk(SkManaged_Png_Chunk_Reader* d, void* context, const char tag[], const void* data, size_t length) {
+    if (!gProcs.fReadChunk) return false;
+    return gProcs.fReadChunk(ToManaged_Png_Chunk_Reader(d), context, tag, data, length);
+}
+
+void destroy(SkManaged_Png_Chunk_Reader* d, void* context) {
+    if (gProcs.fDestroy) {
+        gProcs.fDestroy(ToManaged_Png_Chunk_Reader(d), context);
+    }
```

Comment:
Literally 1 line less... Yay! 😆 

```suggestion
    if (gProcs.fDestroy) return;
    gProcs.fDestroy(ToManaged_Png_Chunk_Reader(d), context);
```

Reply:
you need to invert that lol

Reply:
```c
    if (!gProcs.fDestroy) return false;
    gProcs.fDestroy(ToManaged_Png_Chunk_Reader(d), context);
```